src/lib/adamant-api/index.js

```diff
@@ -148,27 +162,54 @@ export function getPublicKey(address = '') {
  * @returns {Promise<{success: boolean, transactionId: string}>}
  */
 export function sendMessage(params) {
-  return getPublicKey(params.to)
-    .then((publicKey) => {
+  let realId;
+
+  const { to, id, amount, type, message } = params;
+
+  return getPublicKey(to)
+    .then(async (publicKey) => {
       const text =
-        typeof params.message === 'string' ? params.message : JSON.stringify(params.message)
+        typeof message === 'string' ? message : JSON.stringify(message)
       const encoded = utils.encodeMessage(text, publicKey, myKeypair.privateKey)
       const chat = {
         message: encoded.message,
         own_message: encoded.nonce,
-        type: params.type || 1
+        type: type || 1
       }
 
       const transaction = newTransaction(Transactions.CHAT_MESSAGE)
-      transaction.amount = params.amount ? utils.prepareAmount(params.amount) : 0
+      transaction.amount = amount ? utils.prepareAmount(amount) : 0
       transaction.asset = { chat }
-      transaction.recipientId = params.to
+      transaction.recipientId = to
 
       return client.post('/api/chats/process', (endpoint) => {
-        return { transaction: signTransaction(transaction, endpoint.timeDelta) }
+        const signedTransaction = signTransaction(transaction, endpoint.timeDelta)
```

Comment:
Move transaction signing outside `sendMessage`, specifically into the Vuex chat module, and pass the signed transaction as an argument to this function. Do not use Vuex in this function (see https://github.com/Adamant-im/adamant-im/pull/738#discussion_r1979843417).